specification/communication/data-plane/Identity/stable/2023-08-01/CommunicationIdentity.json

```diff
@@ -0,0 +1,470 @@
+            "type": "string",
+            "format": "uuid"
+          },
+          {
+            "in": "header",
+            "name": "Repeatability-First-Sent",
+            "description": "If Repeatability-Request-ID header is specified, then Repeatability-First-Sent header must also be specified. The value should be the date and time at which the request was first created, expressed using the IMF-fixdate form of HTTP-date",
+            "type": "string",
+            "format": "date-time"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Request payload for the token exchange.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/TeamsUserExchangeTokenRequest"
+            }
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "../../../Common/stable/2022-07-13/common.json#/definitions/CommunicationErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/CommunicationIdentityAccessToken"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Exchange a Teams user access token with optional idempotent processing using Repeatability-Request-ID and Repeatability-First-Sent headers.If the headers are not provided, the request will be processed in a non-idempotent way.": {
+            "$ref": "./examples/ExchangeTeamsUserAccessToken.json"
+          }
+        }
+      }
+    },
+    "/identities/{id}/:issueAccessToken": {
+      "post": {
+        "tags": [
+          "Token"
+        ],
+        "summary": "Issue a new token for an identity.",
+        "operationId": "CommunicationIdentity_IssueAccessToken",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "id",
+            "description": "Identifier of the identity to issue token for.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "header",
+            "name": "Repeatability-Request-ID",
+            "description": "If specified, the client directs that the request is repeatable; that is, that the client can make the request multiple times with the same Repeatability-Request-Id and get back an appropriate response without the server executing the request multiple times. The value of the Repeatability-Request-Id  is an opaque string representing a client-generated, globally unique for all time, identifier for the request. It is recommended to use version 4 (random) UUIDs.",
```

Comment:
```suggestion
            "description": "If specified, the client directs that the request is repeatable; that is, that the client can make the request multiple times with the same Repeatability-Request-Id and get back an appropriate response without the server executing the request multiple times. The value of the Repeatability-Request-Id is an opaque string representing a client-generated, globally unique for all time, identifier for the request. It is recommended to use version 4 (random) UUIDs.",
```

Reply:
can you also highlight the fact that Contoso should not use any of their internal identifiers when issuing the idempotency identifier (Repeatability-Request-ID). it really needs to be an opaque and meaningless string.

Reply:
Ok. I added the formulation.